Launcher/Program.cs

```diff
@@ -0,0 +1,34 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Threading;
+using QuantConnect;
+using QuantConnect.Lean.Engine;
+
+namespace Launcher
+{
+    class Program
+    {
+        /// <summary>
+        /// Primary Analysis Thread:
+        /// </summary>
+        public static void Main(string[] args)
+        {
+            QuantConnect.Lean.Engine.Engine.Launch();
```

Comment:
Engine should take its configuration items in its constructor. This is what I meant by the "var config ...." line. Its actually the LeanEngineSystemHandlers etc. Basically everything in the Main() method can be moved into the Launcher (i.e. not duplicated, but moved from the Engine.Main()).